pkg/metrics/metric_sample.go

```diff
@@ -115,9 +117,9 @@ func (m *MetricSample) GetHost() string {
 }
 
 // GetTags returns the metric sample tags
-func (m *MetricSample) GetTags(taggerBuffer, metricBuffer tagset.TagsAccumulator) {
+func (m *MetricSample) GetTags(taggerBuffer, metricBuffer tagset.TagsAccumulator, fn EnrichTagsfn) {
```

Comment:
I'm not sure passing a function pointer is the best pattern here, I think we can fix this function in a better way: 

1. `// GetTags returns the metric sample tags` is no longer accurate, this function doesn't return anything (this was changed long ago), but we can update it. 

2. instead of passing a function pointer, since it depends on public properties of this struct, you can call `tagger.EnrichTags(taggerBuffer, m.OriginFromUDS, m.OriginFromClient` at the call site of `GetTags` [here](https://github.com/DataDog/datadog-agent/blob/4bde6f83b9055afce320a3a73e8a85c0bbb4db55/pkg/aggregator/no_aggregation_stream_worker.go#L195) and [here](https://github.com/DataDog/datadog-agent/blob/4bde6f83b9055afce320a3a73e8a85c0bbb4db55/pkg/aggregator/context_resolver.go#L72)

This way, `MetricSample` no longer directly depends on the tagger, and we can remove the function pointer. 

WDYT?

Reply:
1. Please suggest wording for `GetTags` - happy to change that. 
2. In both the places we use `MetricSampleContext` interface that doesn't have access to m.OriginFromUDS, m.OriginFromClient. It wouldn't be possible to call tagger on the site of `GetTags`.

Reply:
I think for 1, we can just update the comment to be accurate. 
maybe `// GetTags adds the tags to the TagsAccumulator`? (open to suggestions)

For 2, you are right, I suppose an alternative is to provide the tagger as a dependency to the `MetricSample` (as an interface) so it's only used in the relevant implementation instead of adding a function parameter to all implementations of `MetricSampleContext`.

Reply:
We spoke offline, my suggestion really isn't much cleaner. We can leave this as is.